src/form-validator.ts

```diff
@@ -0,0 +1,146 @@
+import { openmrsFetch } from "@openmrs/esm-framework";
+import { async } from "rxjs";
+
+export const handleFormValidation = async (schema) => {
```

Comment:
Ideally, this function should just return already processed and classified validation results. As part of the results, I would only expect errors and warnings classified in an explicit way.